Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -317,6 +336,23 @@ def test_module(client: Client, *_) -> tuple[str, dict[Any, Any], list[Any]]:
     return msg if msg else 'ok', {}, []
 
 
+def result_to_list_of_dicts(client: Client, result: list[dict], headers: list[str]) -> list[dict]:
+    """
+    This function pre-processes the query's result to a list of dictionaries.
+    """
+    if client.dialect == TERADATA:
+        # binding the headers with the columns
+        converted_table = [dict(zip(headers, row)) for row in result]
+    else:
+        # converting a sqlalchemy object to a table
+        converted_table = [dict(row) for row in result]
+
+    # converting b'' and datetime objects to readable ones
+    table = [{str(key): str(value) for key, value in dictionary.items()} for dictionary in converted_table]
```

Comment:
Are we sure this won't break complex values (binary, timestamp) in existing instances on update?

Reply:
Yes, we sure. This line is not a new line, just in a new place in the code.